swagger.yaml

```diff
@@ -1,83 +1,224 @@
+            $ref: "#/definitions/Errors"
+      security:
+        - locations_api_key: []
+
+  /locations/{buildingID}:
+    get:
+      tags:
+        - locations
+      description: "Finds location by buildingID"
+      operationId: getLocationByBuildingID
+      produces:
+        - application/json
       parameters:
-        - name: message
-          in: body
-          schema:
-            type: string
+        - name: buildingID
+          in: path
+          description: "buildingID of location to be searched by"
           required: true
+          type: string
       responses:
-        200:
-          description: The message received
+        "200":
+          description: "Successful response"
+          schema:
+            $ref: "#/definitions/ResultObject"
+        "404":
+          description: "Resource not found"
           schema:
-            type: string
-        404:
-          description: Not found
+            $ref: "#/definitions/Errors"
+        "500":
+          description: "Internal Server Error"
           schema:
-            $ref: '#/definitions/Error'
+            $ref: "#/definitions/Errors"
+      security:
+        - locations_api_key: []
+    
+
+securityDefinitions:
+
+  locations_api_key:
+    type: apiKey
+    name: locations_api_key
+    in: query
+
 definitions:
-  Info:
+  ResultObject:
+    properties:
+      links:
+        $ref: "#/definitions/Links"
+      data:
+        $ref: "#/definitions/ResourceObject"
+
+  ResultObjects:
+    properties:
+      links:
+        $ref: "#/definitions/Links"
+      data:
+        type: array
+        items:
+          $ref: "#/definitions/ResourceObject"
+        
+  ResourceObject:
+    properties:
+      id:
+        type: string
+      type:
+        type: string
+      attributes:
+        $ref: "#/definitions/Attributes"
+        
+  Attributes:
     properties:
       name:
         type: string
-      time:
-        type: integer
-      commit:
+      abbreviation:
         type: string
-      documentation:
+      latitude:
         type: string
-  Error:
+      longitude:
+        type: string        
+      summary:
+        type: string
+      description:
+        type: string
+      address:
+        type: string
+      city:
+        type: string
+      state:
+        type: string
+      zip:
+        type: string
+      telephone:
+        type: string
+      fax:
+        type: string
+      thumbnails:
+        type: array
+        items:
+          type: string
+      images:
+        type: array
+        items:
+          type: string
+      departments:
+        type: array
+        items:
+          type: string
+      website:
+        type: string
+      sqft:
+        type: string
+      calendar:
+        type: string
+      campus:
+        type: string
+      type:
+        type: string
+      open_hours:
+        type: array
+        items:
+          $ref: "#/definitions/DayHours"
+
+  DayHours:
+    properties:
+      day:
```

Comment:
[swagger editor](http://editor.swagger.io/#/) gives the error `Data does not match any schemas from 'anyOf'`